azure-core/azure/core/pipeline/policies/universal.py

```diff
@@ -0,0 +1,310 @@
+                return ET.fromstring(data_as_str)
+            except ET.ParseError:
+                # It might be because the server has an issue, and returned JSON with
+                # content-type XML....
+                # So let's try a JSON load, and if it's still broken
+                # let's flow the initial exception
+                def _json_attemp(data):
+                    try:
+                        return True, json.loads(data)
+                    except ValueError:
+                        return False, None # Don't care about this one
+                success, json_result = _json_attemp(data)
+                if success:
+                    return json_result
+                # If i'm here, it's not JSON, it's not XML, let's scream
+                # and raise the last context in this block (the XML exception)
+                # The function hack is because Py2.7 messes up with exception
+                # context otherwise.
+                _LOGGER.critical("Wasn't XML not JSON, failing")
+                raise_with_traceback(DecodeError, "XML is invalid")
+        raise DecodeError("Cannot deserialize content-type: {}".format(content_type))
+
+    @classmethod
+    def deserialize_from_http_generics(cls, body_bytes, headers):
+        # type: (Optional[Union[AnyStr, IO]], Mapping) -> Any
+        """Deserialize from HTTP response.
+
+        Use bytes and headers to NOT use any requests/aiohttp or whatever
+        specific implementation.
+        Headers will tested for "content-type"
+        """
+        # Try to use content-type from headers if available
+        content_type = None
+        if 'content-type' in headers:
+            content_type = headers['content-type'].split(";")[0].strip().lower()
+        # Ouch, this server did not declare what it sent...
+        # Let's guess it's JSON...
+        # Also, since Autorest was considering that an empty body was a valid JSON,
+        # need that test as well....
+        else:
+            content_type = "application/json"
+
+        if body_bytes:
+            return cls.deserialize_from_text(body_bytes, content_type)
+        return None
+
+    def on_response(self, request, response, **kwargs):
+        # type: (PipelineRequest, PipelineResponse, Any) -> None
+        """Extract data from the body of a REST response object.
+
+        This will load the entire payload in memory.
+
+        Will follow Content-Type to parse.
+        We assume everything is UTF8 (BOM acceptable).
+
+        :param raw_data: Data to be processed.
+        :param content_type: How to parse if raw_data is a string/bytes.
+        :raises JSONDecodeError: If JSON is requested and parsing is impossible.
+        :raises UnicodeDecodeError: If bytes is not UTF8
+        :raises xml.etree.ElementTree.ParseError: If bytes is not valid XML
+        """
+        # If response was asked as stream, do NOT read anything and quit now
+        if kwargs.get("stream", True):
```

Comment:
Is the default really `stream=True`? I thought receiving the data as a stream was opt-in....

Reply:
That's an excellent question:
- old autorest assumed stream=True, so to be backward compatible, with no information I have to say True
- starting Autorest v2.5 ish, I ALWAYS generate "stream=False" explicitly
- in async code, I assume False (because I want to so badly this to be my default)

That being said, it seems there is a push for lazy loading of body, if that's the case, stream=True is the right default :)